### PR TITLE
CB-18484 Modify proxy config of DataLake

### PR DESCRIPTION
**Pull request by Bajzathd**
Datalake service tracks the progress of the underlying stack's proxy config modification.
The actual modification in core service consists of the following steps:
 - Update salt state
 - Apply the modifyproxy salt state to update the proxy settings on the nodes
 - Update CM's proxy settings
 - Update the userdata

### CODE REVIEW

File: datalake/src/test/java/com/sequenceiq/datalake/flow/modifyproxy/handler/ModifyProxyConfigWaitHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+import com.sequenceiq.cloudbreak.common.event.Selectable;
+import com.sequenceiq.cloudbreak.eventbus.Event;
+import com.sequenceiq.datalake.flow.modifyproxy.event.ModifyProxyConfigSuccessResponse;
+import com.sequenceiq.datalake.flow.modifyproxy.event.ModifyProxyConfigWaitRequest;
+import com.sequenceiq.datalake.service.sdx.PollingConfig;
+import com.sequenceiq.datalake.service.sdx.flowwait.SdxWaitService;
+import com.sequenceiq.flow.reactor.api.handler.HandlerEvent;
+
+@ExtendWith(MockitoExtension.class)
+class ModifyProxyConfigWaitHandlerTest {
```
Comment (bergerdenes) [inline datalake/src/test/java/com/sequenceiq/datalake/flow/modifyproxy/handler/ModifyProxyConfigWaitHandlerTest.java:28]: No test for the failure case.

File: environment/src/test/java/com/sequenceiq/environment/environment/service/sdx/SdxServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        when(sdxInternalEndpoint.modifyProxy(SDX_CRN, PREVIOUS_PROXY_CRN, CURRENT_USER_CRN)).thenThrow(cause);
+
+        assertThatThrownBy(() -> doAsCurrentUserCrn(() -> underTest.modifyProxy(SDX_CRN, PREVIOUS_PROXY_CRN)))
+                .isInstanceOf(SdxOperationFailedException.class)
+                .hasCause(cause)
+                .hasMessage(EXTRACTED_MESSAGE);
+        verify(webApplicationExceptionMessageExtractor).getErrorMessage(cause);
+    }
+
+    private static <T> T doAsCurrentUserCrn(Supplier<T> callable) {
```
Comment (bergerdenes) [inline environment/src/test/java/com/sequenceiq/environment/environment/service/sdx/SdxServiceTest.java:100]: Static is not needed

**Review by bergerdenes — CHANGES_REQUESTED** [review]
Minor things

**Review by bergerdenes — APPROVED** [review]
LGTM

File: core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/modifyproxy/ModifyProxyConfigStatusService.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    public void success(Long stackId) {
+        LOGGER.info("Finished modify proxy config state");
+        stackUpdater.updateStackStatus(stackId, DetailedStackStatus.AVAILABLE);
+        flowMessageService.fireEventAndLog(stackId, UPDATE_IN_PROGRESS.name(), CLUSTER_MODIFY_PROXY_CONFIG_SUCCESS);
+    }
+
+    public void failed(Long stackId, Exception exception) {
+        LOGGER.info("Failed modify proxy config state", exception);
```
Comment (lacikaaa) [inline core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/modifyproxy/ModifyProxyConfigStatusService.java:50]: this should be error

File: core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/modifyproxy/action/ModifyProxyConfigAction.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    private ModifyProxyConfigStatusService modifyProxyConfigStatusService;
+
+    protected ModifyProxyConfigAction(Class<P> payloadClass) {
+        super(payloadClass);
+    }
+
+    @Override
+    protected ModifyProxyConfigContext createFlowContext(FlowParameters flowParameters,
+            StateContext<ModifyProxyConfigState, ModifyProxyConfigEvent> stateContext, P payload) {
+        StackView stackView = stackDtoService.getStackViewById(payload.getResourceId());
```
Comment (lacikaaa) [inline core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/modifyproxy/action/ModifyProxyConfigAction.java:38]: we usually set up the mdccontext at this point just to be sure we have it

File: datalake/src/main/java/com/sequenceiq/datalake/controller/sdx/SdxInternalController.java [hunk 5]
```diff
@@ -63,4 +64,11 @@ public void updateDbEngineVersion(@TenantAwareParam String crn, String dbEngineV
         MDCBuilder.addResourceCrn(crn);
         sdxService.updateDatabaseEngineVersion(crn, dbEngineVersion);
     }
+
+    @Override
+    @InternalOnly
+    public FlowIdentifier modifyProxy(@TenantAwareParam String crn, String previousProxyCrn, @InitiatorUserCrn String initiatorUserCrn) {
```
[the remark this answers is not in the document]
Comment (lacikaaa) [reply datalake/src/main/java/com/sequenceiq/datalake/controller/sdx/SdxInternalController.java:70]: I see it's crn everywhere else, so maybe it's obvious for others :)

**Review by lacikaaa — APPROVED** [review]

**Comment by sodre90** [comment]
wouldn't be nice to create an e2e test for this feature or do we have jira ticket for this?

**Review by sodre90 — APPROVED** [review]

**Pull request merged**